crates/ruff_python_formatter/src/pattern/pattern_match_as.rs

```diff
@@ -1,19 +1,42 @@
 use ruff_formatter::{write, Buffer, FormatResult};
-use ruff_python_ast::PatternMatchAs;
+use ruff_python_ast::{Pattern, PatternMatchAs};
 
-use crate::{not_yet_implemented_custom_text, FormatNodeRule, PyFormatter};
+use crate::expression::parentheses::parenthesized;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
 
 #[derive(Default)]
 pub struct FormatPatternMatchAs;
 
 impl FormatNodeRule<PatternMatchAs> for FormatPatternMatchAs {
     fn fmt_fields(&self, item: &PatternMatchAs, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(
-            f,
-            [not_yet_implemented_custom_text(
-                "x as NOT_YET_IMPLEMENTED_PatternMatchAs",
-                item
-            )]
-        )
+        let PatternMatchAs {
+            range: _,
+            pattern,
+            name,
+        } = item;
+
+        if let Some(name) = name {
+            if let Some(pattern) = pattern {
+                // Parenthesize nested `PatternMatchAs` like `(a as b) as c`.
+                if matches!(
+                    pattern.as_ref(),
+                    Pattern::MatchAs(PatternMatchAs {
+                        pattern: Some(_),
+                        ..
+                    })
+                ) {
+                    parenthesized("(", &pattern.format(), ")").fmt(f)?;
```

Comment:
I expect that Black's logic is to preserve parentheses around nested patterns. But we can track this in a separate issue